File: challenge8/breaking-best-and-worst-records.js
function breakingRecords(scores) {
    // Write your code here
    let min = max = scores[0]
    let minCount = maxCount = 0
    if(scores.length === 1) return [0, 0];
    for (let i = 1; i < scores.length; i++) {
        if(scores[i] < min) {
            min = scores[i]
            minCount++
        }
        if(scores[i] > max) {
            max = scores[i]
            maxCount++
        }
    }
    return [maxCount, minCount]
}

console.log(breakingRecords([10, 5, 20, 20, 4, 5, 2, 25, 1])) // [2, 4]
console.log(breakingRecords([3, 4, 21, 36, 10, 28, 35, 5, 24, 42])) // [4, 0]

File: challenge8/breakingBestAndWorstRecords.py
def breakingRecords(scores: list[int]):
    # Write your code here
    min = max = scores[0]
    minCount = maxCount = 0
    if len(scores) == 0:
        return [0, 0]
    for score in scores[1:]:
        if score > max:
            max = score
            maxCount += 1
        elif score < min:
            min = score
            minCount += 1
    return [maxCount, minCount]

if __name__ == '__main__':
    print(breakingRecords([10, 5, 20, 20, 4, 5, 2, 25, 1])) # [2, 4]
    print(breakingRecords([3, 4, 21, 36, 10, 28, 35, 5, 24, 42])) # [4, 0]
